src/btree/bt_cursor.c

```diff
@@ -825,6 +831,7 @@ __wt_btcur_insert(WT_CURSOR_BTREE *cbt)
      *
      * Fixed-length column store can never use a positioned cursor to update because the cursor may
      * not be positioned to the correct record in the case of implicit records in the append list.
+     * FUTURE: it appears that this is no longer true...
```

Comment:
The __cursor_fix_implicit stuff makes it so you can position on any appended recno. And col_modify only relies on being passed the proper recno (which comes out of cbt->iface.recno) and will DTRT if you modify a recno for which there's no update chain yet. So I'm pretty sure that nowadays it will all work. I have no idea what of that was different back when this workaround was added... something must have been...

Reply:
I'm pretty sure FLCS broke when I added the pinned support, I think it's unlikely I would have proactively thought to exclude FLCS. Can you make checking pinned FLCS pages another item for your list, probably simpler to make it another ticket.

Reply:
It already is; we talked about it briefly a while back. The original ticket from when it broke is WT-4267, and I'm pretty sure you found that, not me :-)